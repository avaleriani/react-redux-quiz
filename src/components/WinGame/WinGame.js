import React from "react";
import Btn from "../Btn/Btn";
import styles from "./styles.module.css";

/**
 * @param {{restartGame: function, score: string}} props
 * @return HTMLElement
 */
const WinGame = (props) => (
  <div className={styles['win']}>
    <div className={styles['win-text']}>
      <span role="img" aria-label="party-emoji">🎉🎉  🏆  🎉🎉</span>
      <div>Congratulations, you Win!</div>
      <span role="img" aria-label="party-emoji">🎉🎉  🏆  🎉🎉</span>
      <br/>
      <div>Your score: {props.score}</div>
    </div>
    <Btn action={props.restartGame} text="Play again"/>

  </div>
);

export default WinGame;