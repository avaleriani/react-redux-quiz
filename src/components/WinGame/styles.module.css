.win {
    display: flex;
    flex-direction: column;
}

.win-text {
    border: 2px solid #74B559;
    padding: 30px;
    text-align: center;
    margin-bottom: 54px;
    font-size: 1.2rem;
}