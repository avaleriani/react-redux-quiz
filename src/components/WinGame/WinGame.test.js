import React from 'react';
import renderer from 'react-test-renderer';
import WinGame from './WinGame';

describe('WinGame component', () => {

  test('renders correctly without props', () => {
    const WinGameComponent = renderer.create(<WinGame/>).toJSON();
    expect(WinGameComponent).toMatchSnapshot();
  });

  test('renders with props', () => {
    const props = {
      score: 55, restartGame: () => {
      }
    };
    const WinGameComponent = renderer.create(<WinGame {...props}/>).toJSON();
    expect(WinGameComponent).toMatchSnapshot();
  });
});