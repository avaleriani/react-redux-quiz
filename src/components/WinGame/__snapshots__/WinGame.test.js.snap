// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinGame component renders correctly without props 1`] = `
<div
  className="win"
>
  <div
    className="win-text"
  >
    <span
      aria-label="party-emoji"
      role="img"
    >
      🎉🎉  🏆  🎉🎉
    </span>
    <div>
      Congratulations, you Win!
    </div>
    <span
      aria-label="party-emoji"
      role="img"
    >
      🎉🎉  🏆  🎉🎉
    </span>
    <br />
    <div>
      Your score: 
    </div>
  </div>
  <button
    className="button"
    onClick={null}
    type="button"
  >
    Play again
  </button>
</div>
`;

exports[`WinGame component renders with props 1`] = `
<div
  className="win"
>
  <div
    className="win-text"
  >
    <span
      aria-label="party-emoji"
      role="img"
    >
      🎉🎉  🏆  🎉🎉
    </span>
    <div>
      Congratulations, you Win!
    </div>
    <span
      aria-label="party-emoji"
      role="img"
    >
      🎉🎉  🏆  🎉🎉
    </span>
    <br />
    <div>
      Your score: 
      55
    </div>
  </div>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Play again
  </button>
</div>
`;
